### ISSUE TITLE
Only show files in the file panel with a truthy url

### ISSUE THREAD

**Issue by turt2live** [vector-im/riot-web#6250]
### Description

As per https://gitlab.gnome.org/danigm/fractal/issues/101

... or to summarize the issue, an event like the following will appear in the file panel:
```
  "content": {
    "body": "Ooh, Fractal has an icon in the Papirus icon pack",
    "url": null,
    "msgtype": "m.text",
    "formatted_body": null,
    "format": null
  },
```

### Version information

- **Platform**: web (in-browser)
- **Browser**: Chrome 64
- **OS**: Windows 10
- **URL**: riot.im/develop

**Comment by uhoreg** [comment]
see also https://github.com/vector-im/riot-web/issues/4480 and https://github.com/vector-im/riot-web/issues/5076.

**Comment by t3chguy** [comment]
events are indexed with whether or not they `contains_url` so this'll be a Synapse bug :(

**Comment by t3chguy** [comment]
https://github.com/matrix-org/synapse/blob/752b7b32ed1c33651c0c64fbbb4289c3b62ac89b/synapse/storage/events.py#L1119-L1122

it'll also need some migration magic to fix this for all existing events in db

**Comment by t3chguy** [comment]
This seems fine, so maybe its a fixed Synapse bug:
![image](https://user-images.githubusercontent.com/2403652/41503208-3b2e7212-71c5-11e8-83ef-e6c93c0c0da7.png)

**Comment by t3chguy** [comment]
weird, even though its right in the db and the filter seems fine the event is in my FilePanel 
![image](https://user-images.githubusercontent.com/2403652/41503230-ce4d5e5a-71c5-11e8-8e61-673a7a68b05a.png)

**Comment by t3chguy** [comment]
ah turns out it is a Synapse bug where the `is_url` "formula" is inconsistent between files

**Comment by t3chguy** [comment]
though as per the spec:
![image](https://user-images.githubusercontent.com/2403652/41507071-ba33c208-7222-11e8-8a89-936cd6657f3d.png)
it won't exclude non-truthy but instead non-string

**Comment by t3chguy** [comment]
This is actually a weird combination of a Synapse bug and a js-sdk one.
So if the event was paginated in then the js-sdk bug would hit and otherwise the Synapse bug would hit.

**Comment by t3chguy** [comment]
though technically the matrix spec here could be understood to mean that `"url": null` should trip this Filter,
Synapse in 2/3 places checked whether it was a string, Dendrite does not. So its a 50/50 right now. I might write a proposal to improve `contains_url` to checking for a truthy string to get around this

**Comment by kittykat** [comment]
@turt2live What would it take to resolve this issue? To do the checks in Dendrite?

**Comment by turt2live** [comment]
I don't think this is a dendrite issue. We should either filter it out locally, or declare it a synapse bug and move it there.
